ledger-tool/src/main.rs

```diff
@@ -486,12 +487,20 @@ fn analyze_storage(database: &Database) -> Result<(), String> {
 }
 
 fn open_genesis_config(ledger_path: &Path) -> GenesisConfig {
-    GenesisConfig::load(&ledger_path).unwrap_or_else(|err| {
-        eprintln!(
-            "Failed to open ledger genesis_config at {:?}: {}",
-            ledger_path, err
+    GenesisConfig::load(&ledger_path).unwrap_or_else(|load_err| {
```

Comment:
As a refactoring bonus, I've supported archive genes in `ledger-tool`, which is handy for testing purpose and for consistency.